Automattic/newspack-plugin
# feat(connections): add MailChimp API key setting in connection screen

### All Submissions:






### Changes proposed in this Pull Request:



https://github.com/Automattic/newspack-popups/issues/341

MailChimp API key is managed in the newsletter plugin while it's used in other plugins too. Move the
MailChimp API key option to the main plugin (this one).

~~BREAKING CHANGE: No breaking changes for this plugin, but for the plugins using the MailChimp API Key option they need to use the new one from this commit instead.~~
I changed back the option name to the one used in `newspack-newsletter` plugin and the others.

### How to test the changes in this Pull Request:

1. Check out this.
2. Define `NEWSPACK_WPCOM_CLIENT_ID` in your `wp-config.php` to enable the `connections` menu (a dummy value should work too as it's not used in this feature). @adekbadek maybe we can remove the condition to show the menu item https://github.com/Automattic/newspack-plugin/blob/5181873cbce1c7ce27aeb1a3896683bef627d510/includes/wizards/class-dashboard.php#L112 or can we add a way to enable it when a publisher is using MailChimp?
3. Click on the `Connections` menu item.
4. Observe the MailChimp card configuration, and click on `connect`.
5. Enter a MailChimp API key, you can get one from the [MailChimp API Dashboard](https://us1.admin.mailchimp.com/account/api/), and click on `Connect`.
6. If the API key is valid, observe that MailChimp is set as connected, and showing the username related to the API key.

### Other information:

~~The branch of this PR is based on the [main connections branch](https://github.com/Automattic/newspack-plugin/tree/feat/connections-view), once the main branch is merged on master, I'll change back the parent branch to master.~~ This is done.
